crates/bevy_core_pipeline/src/dof/mod.rs

```diff
@@ -0,0 +1,907 @@
+                pad_a: 0,
+                pad_b: 0,
+                pad_c: 0,
+            },
+        ));
+    }
+}
+
+/// Given the sensor height and the FOV, returns the focal length.
+///
+/// See <https://photo.stackexchange.com/a/97218>.
+pub fn calculate_focal_length(sensor_height: f32, fov: f32) -> f32 {
+    0.5 * sensor_height / f32::tan(0.5 * fov)
+}
+
+impl DepthOfFieldPipelines {
+    /// Populates the information that the `DepthOfFieldNode` needs for the two
+    /// depth of field render passes.
+    fn pipeline_render_info(&self) -> [DepthOfFieldPipelineRenderInfo; 2] {
+        match *self {
+            DepthOfFieldPipelines::Gaussian {
+                horizontal: horizontal_pipeline,
+                vertical: vertical_pipeline,
+            } => [
+                DepthOfFieldPipelineRenderInfo {
+                    pass_label: "depth of field pass (horizontal Gaussian)",
+                    view_bind_group_label: "depth of field view bind group (horizontal Gaussian)",
+                    pipeline: horizontal_pipeline,
+                    is_dual_input: false,
+                    is_dual_output: false,
+                },
+                DepthOfFieldPipelineRenderInfo {
+                    pass_label: "depth of field pass (vertical Gaussian)",
+                    view_bind_group_label: "depth of field view bind group (vertical Gaussian)",
+                    pipeline: vertical_pipeline,
+                    is_dual_input: false,
+                    is_dual_output: false,
+                },
+            ],
+
+            DepthOfFieldPipelines::Bokeh {
+                pass_0: pass_0_pipeline,
+                pass_1: pass_1_pipeline,
+            } => [
+                DepthOfFieldPipelineRenderInfo {
+                    pass_label: "depth of field pass (bokeh pass 0)",
+                    view_bind_group_label: "depth of field view bind group (bokeh pass 0)",
+                    pipeline: pass_0_pipeline,
+                    is_dual_input: false,
+                    is_dual_output: true,
+                },
+                DepthOfFieldPipelineRenderInfo {
+                    pass_label: "depth of field pass (bokeh pass 1)",
+                    view_bind_group_label: "depth of field view bind group (bokeh pass 1)",
+                    pipeline: pass_1_pipeline,
+                    is_dual_input: true,
+                    is_dual_output: false,
+                },
+            ],
+        }
+    }
+}
+
+/// Returns true if multisampled depth textures are supported on this platform.
+///
+/// In theory, Naga supports depth textures on WebGL 2. In practice, it doesn't,
+/// because of a silly bug whereby Naga assumes that all depth textures are
+/// `sampler2DShadow` and will cheerfully generate invalid GLSL that tries to
+/// perform non-percentage-closer-filtering with such a sampler. Therefore we
+/// disable depth of field entirely on WebGL 2.
+#[cfg(target_arch = "wasm32")]
```

Comment:
I believe this should just check for webgl, since webgpu should support depth textures. This currently disables it for webgpu too.

Reply:
Incantation for that is `#[cfg(all(feature = "webgl", target_arch = "wasm32", not(feature = "webgpu")))]`

Reply:
couldn't that just be `#[cfg(feature = "webgl")]`?

Reply:
No, because that's enabled by default and you can have both the webgpu and webgl feature enabled at the same time (it targets webgpu if both are enabled, to avoid having mutually exclusive features).